python/grass/grassdb/history.py

```diff
@@ -18,104 +19,343 @@
 def get_current_mapset_gui_history_path():
     """Return path to the current mapset history file."""
     env = gisenv()
-    return os.path.join(
-        env["GISDBASE"], env["LOCATION_NAME"], env["MAPSET"], ".wxgui_history"
-    )
+    base_path = os.path.join(env["GISDBASE"], env["LOCATION_NAME"], env["MAPSET"])
+    history_filename = ".wxgui_history"
+
+    txt_path = os.path.join(base_path, history_filename)
+    json_path = os.path.join(base_path, f"{history_filename}.json")
+
+    # Return path to plain-text file if exists otherwise return JSON
+    return txt_path if os.path.exists(txt_path) else json_path
+
+
+def get_extension():
+    """Return extension of the current mapset history file.
+    The standard long-term format is plain text.
+    Newly it is stored as JSON-formatted.
+    """
+    history_path = get_current_mapset_gui_history_path()
+    file_path, file_name = os.path.split(history_path)
+    file_name_no_ext, extension = os.path.splitext(file_name)
+    return extension
```

Comment:
Let's use pathlib for this.

Reply:
I used pathlib in the whole file, it is clearer.